pkg/provider/aws/execution_unit_test.go

```diff
@@ -0,0 +1,182 @@
+				ExecutionUnits: map[string]*config.ExecutionUnit{
+					"test": &config.ExecutionUnit{Type: "lambda"},
+				},
+			},
+			want: testResult{
+				nodes: []core.Resource{
+					repo, image, role, lambda,
+				},
+				deps: []graph.Edge[core.Resource]{
+					graph.Edge[core.Resource]{
+						Source:      repo,
+						Destination: image,
+					},
+					graph.Edge[core.Resource]{
+						Source:      image,
+						Destination: lambda,
+					},
+					graph.Edge[core.Resource]{
+						Source:      role,
+						Destination: lambda,
+					},
+				},
+			},
+		},
+		{
+			name: "ecr repo already exists",
+			cfg: config.Application{
+				AppName: "test",
+				ExecutionUnits: map[string]*config.ExecutionUnit{
+					"test": &config.ExecutionUnit{Type: "lambda"},
+				},
+			},
+			existingResources: []core.Resource{ecr.NewEcrRepository("test", core.AnnotationKey{ID: "test2", Capability: annotation.ExecutionUnitCapability})},
+			want: testResult{
+				nodes: []core.Resource{
+					repo, image, role, lambda,
+				},
+				deps: []graph.Edge[core.Resource]{
+					graph.Edge[core.Resource]{
+						Source:      repo,
+						Destination: image,
+					},
+					graph.Edge[core.Resource]{
+						Source:      image,
+						Destination: lambda,
+					},
+					graph.Edge[core.Resource]{
+						Source:      role,
+						Destination: lambda,
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			assert := assert.New(t)
+			aws := AWS{
+				Config: &tt.cfg,
+			}
+			dag := core.NewResourceGraph()
+
+			for _, res := range tt.existingResources {
+				dag.AddResource(res)
+			}
+
+			err := aws.GenerateExecUnitResources(unit, dag)
+			if tt.want.err {
+				assert.Error(err)
+				return
+			}
+			if !assert.NoError(err) {
+				return
+			}
+			for _, node := range tt.want.nodes {
+				found := false
+				for _, res := range dag.ListConstructs() {
+					if res.Id() == node.Id() {
+						found = true
+					}
+				}
+				assert.True(found)
+			}
+
+			for _, dep := range tt.want.deps {
+				found := false
+				for _, res := range dag.ListDependencies() {
+					if res.Source.Id() == dep.Source.Id() && res.Destination.Id() == dep.Destination.Id() {
+						found = true
+					}
+				}
+				assert.True(found)
+			}
+
+			for _, res := range dag.ListConstructs() {
+				if repo, ok := res.(*ecr.EcrRepository); ok {
+					if len(tt.existingResources) != 0 {
+						assert.Len(repo.ConstructsRef, 2)
+					} else {
+						assert.Len(repo.ConstructsRef, 1)
+					}
+				}
+			}
+		})
+
+	}
+}
+
+func Test_GetAssumeRolePolicyForType(t *testing.T) {
+	cases := []struct {
+		name  string
+		cfg   config.ExecutionUnit
+		wants string
+	}{
+		{
+			name:  `lambda`,
+			cfg:   config.ExecutionUnit{Type: Lambda},
+			wants: "{\n\tVersion: '2012-10-17',\n\tStatement: [\n\t\t{\n\t\t\tAction: 'sts:AssumeRole',\n\t\t\tPrincipal: {\n\t\t\t\tService: 'lambda.amazonaws.com',\n\t\t\t},\n\t\t\tEffect: 'Allow',\n\t\t\tSid: '',\n\t\t},\n\t],\n}",
```

Comment:
a backticked-string (so it can be multiline, etc) might make these more readable